### PR TITLE
Feat/new topic creation

### PR DESCRIPTION
**Pull request by DonatasTamosauskas**
- Implemented new topic creation.

### CODE REVIEW

File: schedulearn-frontend/src/components/NestedList/TopicAddModal.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  topic?: TopicForm;
+  onRequestClose: () => void;
+  onEventSubmit: (topic: TopicForm) => void;
+}
+
+export class TopicAddModal extends React.PureComponent<TopicAddModalProps> {
+  render(): JSX.Element {
+    return (
+      <Modal
+        isOpen={this.props.isOpen}
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/NestedList/TopicAddModal.tsx:21]: Needs to set portalClassName to ".ReactModal__Portal topic-add-modal" so the style could be applied specifically for your ReactModal. Check andrius Info modal.

File: schedulearn-frontend/src/components/NestedList/TopicAddModal.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.ReactModal__Content {
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/NestedList/TopicAddModal.scss]: .topic-add-modal .ReactModal__Content selector, to apply specifically to your modal, not application wide.

File: schedulearn-frontend/src/components/NestedList/TopicAddForm.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      name: this.props.newTopic?.name ?? "",
+      description: this.props.newTopic?.description ?? "",
+      parentTopicId: this.props.newTopic?.parentTopicId,
+    };
+  }
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    
+    if(this.state.name === "") {
```
Comment (AinoraZ) [inline schedulearn-frontend/src/components/NestedList/TopicAddForm.tsx]: Set the name as required (ex: <input required={true} ... />) instead.

**Review by AinoraZ — CHANGES_REQUESTED** [review]

File: schedulearn-frontend/src/components/NestedList/TopicAddModal.tsx (hunk 1 above)
Comment (DonatasTamosauskas) [reply schedulearn-frontend/src/components/NestedList/TopicAddModal.tsx:21]: `portalClassName` set to `topic-add-modal ReactModalPortal`

**Review by AinoraZ — APPROVED** [review]

**Comment by AinoraZ** [comment]
[User should be able to create new topics](https://trello.com/c/ZS1sPQcg/9-user-should-be-able-to-create-new-topics)

**Pull request merged**